rclcpp/include/rclcpp/strategies/allocator_memory_strategy.hpp

```diff
@@ -61,24 +62,24 @@ class AllocatorMemoryStrategy : public memory_strategy::MemoryStrategy
     allocator_ = std::make_shared<VoidAlloc>();
   }
 
-  void add_guard_condition(const rcl_guard_condition_t * guard_condition) override
+  void add_guard_condition(const rclcpp::GuardCondition & guard_condition) override
   {
-    for (const auto & existing_guard_condition : guard_conditions_) {
-      if (existing_guard_condition == guard_condition) {
-        return;
-      }
-    }
-    guard_conditions_.push_back(guard_condition);
+    add_guard_condition(&guard_condition);
   }
 
-  void remove_guard_condition(const rcl_guard_condition_t * guard_condition) override
+  void add_guard_condition(const rclcpp::GuardCondition::SharedPtr guard_condition) override
   {
-    for (auto it = guard_conditions_.begin(); it != guard_conditions_.end(); ++it) {
-      if (*it == guard_condition) {
-        guard_conditions_.erase(it);
-        break;
-      }
-    }
+    add_guard_condition(guard_condition.get());
```

Comment:
Since this method is not retaining the ownership of the shared pointer (or otherwise using it), can we just get rid of these signatures that take the shared pointer and leave just the ones that take a reference?

Reply:
Done in 7a4b2a1